debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.Requirements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.Envelope;
+import io.debezium.time.MicroTimestamp;
+import io.debezium.time.NanoTimestamp;
+import io.debezium.time.Timestamp;
+import io.debezium.time.ZonedTime;
+import io.debezium.time.ZonedTimestamp;
+
+public class TimezoneConverter<R extends ConnectRecord<R>> implements Transformation<R> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimezoneConverter.class);
+
+    private static final Field CONVERTED_TIMEZONE = Field.create("converted.timezone")
+            .withDisplayName("Converted Timezone")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withValidation(Field::isRequired)
+            .withDescription("A string of time shift value representing timezone in the format of +/-hh:mm, e.g. -08:00 or +05:30.");
```

Comment:
I've alluded to this in my other comment; however, I'm concerned here that the use of `+/-HH:MM` has the potential to be confusing to users who deal with daylight savings time.  For example, if I am configuring this during the summer months and live in Central Europe, do I configure this as `+01:00` still, which represents my offset during the winter, or do I use `+02:00`, which is the current summer offset, or vice versa.

This seems ambiguous, so we'll need to be explicit in the documentation.   My concern is that no matter how clear we make this in the documentation, it will likely be a detail that is overlooked by users and could lead to a series of questions every time DST starts.